src/porepy/models/contact_mechanics_model.py

```diff
@@ -399,6 +399,43 @@ def reconstruct_local_displacement_jump(self, data_edge, from_iterate=True):
         u_mortar_local = project_to_local * displacement_jump_global_coord
         return u_mortar_local.reshape((self.Nd, -1), order="F")
 
+    def reconstruct_stress(self) -> None:
+        """
+        Compute the stress in the highest-dimensional grid based on the displacement
+        states in that grid, adjacent interfaces and global boundary conditions.
+        
+        The stress is stored in the data dictionary of the highest dimensional grid,
+        in [pp.STATE]['stress'].
+        
+        """
+        g = self._nd_grid()
+        d = self.gb.node_props(g)
+
+        mpsa = pp.Mpsa(self.mechanics_parameter_key)
+
+        u = d[pp.STATE][self.displacement_variable]
+
+        matrix_dictionary = d[pp.DISCRETIZATION_MATRICES][self.mechanics_parameter_key]
+
+        # Stress contribution from internal cell center displacements
+        stress = matrix_dictionary[mpsa.stress_matrix_key] * u
+
+        # Contributions from global boundary conditions
+        bound_stress_discr = matrix_dictionary[mpsa.bound_stress_matrix_key]
+        global_bc_val = d[pp.PARAMETERS][self.mechanics_parameter_key]["bc_values"]
+        stress += bound_stress_discr * global_bc_val
+
+        # Contributions from the mortar displacement variables
+        for e, d_e in self.gb:
+            # Only contributions from interfaces to the highest dimensional grid
+            mg = d_e["mortar_grid"]
+            if mg.dim == self.Nd - 1:
+                u_e = d_e[pp.STATE][self.mortar_displacement_variable]
+
+                stress += bound_stress_discr * mg.mortar_to_master_avg(nd=self.Nd) * u_e
+
+        d[pp.STATE]["stress"] = stress
```

Comment:
```suggestion
        d[pp.STATE][self.stress_key] = stress
```
Or stress_reconstruction_key or similar.